### PR TITLE
Forecast `start_timestamp` for CLI command

### LINKED ISSUE #1247: Forecast `start_timestamp` for CLI command
### 🚀 Feature Request

Set forecasting intervall by adding `start_timestamp` parameter.

### Proposal

Add to `forecast` CLI command config the possibility to set forecast interval
1. Compute a new forecast horizon if `start_timestamp` set in  `forecast_params_config`
   General idea of how horizon can be extended:
    ```
    extended_horizon = int(start_timestamp - train_end_timestamp) + horizon
    ```
   When new horizon estimated, timestamp frequency should be taken into account. (weekly, daily, hourly, ...)
3. Estimate pipeline with the new horizon and make predictions
4. Filter forecasts with `timestamp >= start_timestamp`
5. Add `start_timestamp` description to config documentation

### Test cases

* Test that `forecast` command works with `start_timestamp` parameter.
* Test that forecast timestamps `>= start_timestamp` and number of points `== horizon`.

### Additional context

[`forecast_params_config`](https://github.com/tinkoff-ai/etna/blob/master/etna/commands/forecast_command.py#L71)

### PR DESCRIPTION
**Pull request by brsnw250**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Closing issues

closes #1247

### CODE REVIEW

File: docs/source/commands.rst [hunk 1]
```diff
@@ -21,6 +21,14 @@ Basic ``forecast`` usage:
             [RAW_OUTPUT]            by default we return only forecast without features [default: False]
             [KNOWN_FUTURE]          list of all known_future columns (regressor columns). If not specified then all exog_columns considered known_future [default: None]
 
+**Forecast config parameters**
+
+* :code:`prediction_interval` - whether to estimate prediction interval for forecast.
+* :code:`quantiles` - levels of prediction distribution. By default 2.5% and 97.5% are taken to form a 95% prediction interval.
+* :code:`n_folds` - number of folds to use in the backtest for prediction interval estimation.
+* :code:`return_components` - whether to estimate forecast components
+* :code:`start_timestamp` - timestamp with the starting point of forecast.
```
Comment (Mr-Geekman) [inline docs/source/commands.rst:30]: I think we should clarify that this parameter is unnecessary. Same for other parameters probably.

File: docs/source/commands.rst [hunk 2]
```diff
@@ -21,6 +21,14 @@ Basic ``forecast`` usage:
             [RAW_OUTPUT]            by default we return only forecast without features [default: False]
             [KNOWN_FUTURE]          list of all known_future columns (regressor columns). If not specified then all exog_columns considered known_future [default: None]
 
+**Forecast config parameters**
+
+* :code:`prediction_interval` - whether to estimate prediction interval for forecast.
+* :code:`quantiles` - levels of prediction distribution. By default 2.5% and 97.5% are taken to form a 95% prediction interval.
+* :code:`n_folds` - number of folds to use in the backtest for prediction interval estimation.
```
Comment (Mr-Geekman) [inline docs/source/commands.rst:28]: May be we should give here a link to the class where this parameters are explained in documentation.

File: etna/commands/forecast_command.py [hunk 3]
```diff
@@ -83,9 +120,17 @@ def forecast(
 
     tsdataset = TSDataset(df=df_timeseries, freq=freq, df_exog=df_exog, known_future=k_f)
 
+    horizon: int = pipeline_configs["horizon"]  # type: ignore
+    horizon = compute_horizon(horizon=horizon, forecast_params=forecast_params, tsdataset=tsdataset)
+    pipeline_configs["horizon"] = horizon  # type: ignore
+
+    forecast_args = drop_additional_forecast_params(forecast_params)
```
Comment (Mr-Geekman) [inline etna/commands/forecast_command.py:127]: It a difficult to understand naming. May be it is better to name it `forecast_call_params` or `forecast_method_params`.

File: tests/test_commands/test_forecast.py [hunk 4]
```diff
@@ -100,3 +106,112 @@ def test_forecast_use_exog_correct(
     pd.testing.assert_series_equal(
         df_output["target"], pd.Series(data=[3.0, 3.0, 3.0], name="target"), check_less_precise=1
     )
+
+
+@pytest.fixture()
```
Comment (Mr-Geekman) [inline tests/test_commands/test_forecast.py:111]: We usually don't add `()` on fixtures.

File: tests/test_commands/test_forecast.py [hunk 5]
```diff
@@ -100,3 +106,112 @@ def test_forecast_use_exog_correct(
...
+        ),
+    ),
+)
+def test_drop_additional_forecast_params(params, expected):
+    result = drop_additional_forecast_params(forecast_params=params)
+    assert result == expected
+
+
+@pytest.mark.parametrize("forecast_params", ({"start_timestamp": "2020-04-09"}, {"start_timestamp": "2019-04-10"}))
+def test_update_horizon_error(example_tsds, forecast_params, pipeline_dummy_config):
```
Comment (Mr-Geekman) [inline tests/test_commands/test_forecast.py:151]: `test_update_horizon_error` -> `test_compute_horizon_error`?

File: tests/test_commands/test_forecast.py [hunk 6]
```diff
@@ -100,3 +106,112 @@ def test_forecast_use_exog_correct(
...
+@pytest.mark.parametrize(
+    "forecast_params,tsdataset_name,expected",
+    (
+        ({"start_timestamp": "2020-04-10"}, "example_tsds", 3),
+        ({"start_timestamp": "2020-04-12"}, "example_tsds", 5),
+        ({"start_timestamp": "2020-02-01 02:00:00"}, "example_tsdf", 4),
+        ({"start_timestamp": "2023-06-01"}, "ms_tsds", 4),
+    ),
+)
+def test_update_horizon(forecast_params, tsdataset_name, expected, request, pipeline_dummy_config):
```
Comment (Mr-Geekman) [inline tests/test_commands/test_forecast.py:167]: `test_update_horizon` -> `test_compute_horizon`?

File: docs/source/commands.rst [hunk 7]
```diff
@@ -43,6 +51,7 @@ Example of forecast params config:
     prediction_interval: true
     quantiles: [0.025, 0.975]
     n_folds: 3
+    start_timestamp: "2020-01-12"
```
[the remark this answers is not in the document]
Comment (brsnw250) [reply docs/source/commands.rst:64]: Added

File: CHANGELOG.md [hunk 8]
```diff
@@ -9,7 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - Notebook `forecast_interpretation.ipynb` with forecast decomposition ([#1220](https://github.com/tinkoff-ai/etna/pull/1220))
 - Exogenous variables shift transform `ExogShiftTransform`([#1254](https://github.com/tinkoff-ai/etna/pull/1254))
--
+- Parameter `start_timestamp` parameter to forecast CLI command ([#1265](https://github.com/tinkoff-ai/etna/pull/1265))
```
Comment (Mr-Geekman) [inline CHANGELOG.md:12]: Probably, the second word "parameter" should be removed.

File: etna/commands/forecast_command.py (hunk 3 above)
Comment (Mr-Geekman) [reply etna/commands/forecast_command.py]: Sorry, I meant variable `forecast_args`, not method name.

Comment (brsnw250) [reply etna/commands/forecast_command.py]: Oh, ok)

**Review by Mr-Geekman — CHANGES_REQUESTED** [review]
Little changes to do.

**Review by Mr-Geekman — APPROVED** [review]

**Pull request merged**